src/hpc/cpu.c

```diff
@@ -0,0 +1,74 @@
+/****************************************************************************
+**
+**  This file is part of GAP, a system for computational discrete algebra.
+**
+**  Copyright of GAP belongs to its developers, whose names are too numerous
+**  to list here. Please refer to the COPYRIGHT file for details.
+**
+**  SPDX-License-Identifier: GPL-2.0-or-later
+*/
+
+#include "hpc/cpu.h"
+
+/****************************************************************************
+**
+*V  SyNumProcessors  . . . . . . . . . . . . . . . . . number of logical CPUs
+**
+*/
+#ifdef NUM_CPUS
+UInt SyNumProcessors = NUM_CPUS;
+#else
+UInt SyNumProcessors = 4;
+#endif
+
+/****************************************************************************
+**
+*F  SyCountProcessors() . . . . . . . . . . . . compute the number of CPUs.
+**
+**  SyCountProcessors() retrieves the number of active logical processors.
+**  In case of Linux, calls get_nprocs().
+**  In case of Windows 7 and higher, calls
+**  GetActiveProcessorCount(ALL_PROCESSOR_GROUPS).
+**  In case of older versions of Windows, tries to load the number of
+**  processors in current processor group via GetSystemInfo with proper
+**  arguments and falls back to the value of 4 in case of an error.
+**  In case of macOS, sysctl is called with HW_AVAILCPU flag.
+**  In case of other operating systems the value of 4 is returned.
+*/
+
+UInt SyCountProcessors(void) {
+#ifdef NUM_CPUS
+    return NUM_CPUS;
+#else
+    #if __linux__
+        return get_nprocs();
```

Comment:
Actually, this file is not generally available on Linux; rather, it is a GNU extension and as such provided by glibc, but not necessarily all libc implementations available on Linux. I just checked, luckily Musl libc *does* provide it (as a legacy wrapper).

To fix this, I see the following alternatives:
1. use a macro check for GNU libc and use that as a guard (least work needed)
2. actually check for the availability of `get_nprocs` (and also `sys/sysinfo.h` !) in `configure.ac` (better, but more work)
3. avoid `get_nprocs` and instead call `sched_getaffinity(getpid(), ...)` and then count the CPUs in the returned CPU set (this is how musl libc implements `get_nprocs`, and for GNU libc it is the final fallback -- for some reason the first thing it tries is to parse `/sys/devices/system/cpu/online`, if that fails it falls back to parsing `/proc/stat`; if that fails it calls `sched_getaffinity`).

My personal preference would be number 2. I do not mean to burden you with this as a requirement for merging your check; so, if you feel inclined to work on this: great! If not, no worries, then I can take care of it later.

Reply:
Actually, to be precise, on musl libc `get_nprocs` is implemented by a simple call to `sysconf(_SC_NPROCESSORS_ONLN)`, and *that* is handled by checking `sched_getaffinity`. 

According to https://stackoverflow.com/a/150971/928031 it works on Linux, Solaris, AIX and Mac OS X >=10.4 ; and I just checked, current FreeBSD and OpenBSD have it. So perhaps that's the easiest way to go about all this?

Reply:
Yes, calling `sysconf(_SC_NPROCESSORS_ONLN)` makes things really easy. Check out the new code, please.